test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs

```diff
@@ -188,6 +188,75 @@ public async Task GivenVariousTypesOfResources_WhenSearchingAcrossAllResourceTyp
+
+            bundle = await Client.SearchAsync("?_type=Patient1,Patient2");
+            Assert.Contains("_type=Patient1,Patient2", bundle.Link[0].Url);
+            outcome = GetOutcome(bundle);
+            ValidateOperationOutcome(expectedDiagnosticsMultipleWrongTypes, expectedIssueSeverities, expectedCodeTypes, outcome);
+
+            bundle = await Client.SearchPostAsync(null, default, ("_type", "Patient1"));
+            Assert.Contains("_type=Patient1", bundle.Link[0].Url);
+            outcome = GetOutcome(bundle);
+            ValidateOperationOutcome(expectedDiagnosticsOneWrongType, expectedIssueSeverities, expectedCodeTypes, outcome);
+            bundle = await Client.SearchPostAsync(null, default, ("_type", "Patient1,Patient2"));
+            Assert.Contains("_type=Patient1,Patient2", bundle.Link[0].Url);
+            outcome = GetOutcome(bundle);
+            ValidateOperationOutcome(expectedDiagnosticsMultipleWrongTypes, expectedIssueSeverities, expectedCodeTypes, outcome);
+
+            static OperationOutcome GetOutcome(Bundle bundle)
+            {
+                var outcomeEnity = bundle.Entry.Where(x => x.Resource.ResourceType == ResourceType.OperationOutcome).FirstOrDefault();
+                Assert.NotNull(outcomeEnity);
+                var outcome = outcomeEnity.Resource as OperationOutcome;
+                Assert.NotNull(outcome);
+                return outcome;
+            }
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenSomeInvalidTypeOfResources_WhenSearchingAcrossAllResourceTypes_ThenSearchHasProperOutcome()
+        {
+            Patient[] patients = await Client.CreateResourcesAsync<Patient>(3);
+            string[] expectedDiagnostics = { string.Format(Core.Resources.InvalidTypeParameter, "'Patient1'") };
+            OperationOutcome.IssueType[] expectedCodeTypes = { OperationOutcome.IssueType.NotSupported };
+            OperationOutcome.IssueSeverity[] expectedIssueSeverities = { OperationOutcome.IssueSeverity.Warning };
+
+            Bundle bundle = await Client.SearchAsync("?_type=Patient,Patient1");
+            Assert.Contains("_type=Patient,Patient1", bundle.Link[0].Url);
+            OperationOutcome outcome = GetOutcome(bundle);
+            ValidateBundle(bundle, patients.AsEnumerable<Resource>().Append(outcome).ToArray());
+            ValidateOperationOutcome(expectedDiagnostics, expectedIssueSeverities, expectedCodeTypes, outcome);
+
+            bundle = await Client.SearchPostAsync(null, default, ("_type", "Patient1,Patient"));
+            Assert.Contains("_type=Patient1,Patient", bundle.Link[0].Url);
+            outcome = GetOutcome(bundle);
+            ValidateBundle(bundle, patients.AsEnumerable<Resource>().Append(outcome).ToArray());
+            ValidateOperationOutcome(expectedDiagnostics, expectedIssueSeverities, expectedCodeTypes, outcome);
+
+            static OperationOutcome GetOutcome(Bundle bundle)
```

Comment:
Since this local method is in multiple test cases could it be moved to be a class method so there isn't duplicate code.